### PR TITLE
fix change nodename error

### PR DESCRIPTION
**Pull request by dcy**
Hi! Please review this pr. 
fix https://github.com/erlio/vernemq/issues/241

### CODE REVIEW

**Comment by dcy** [comment]
I create another branch, maybe it is better to fix it in ```vmq_subscriber_db.erl```.
https://github.com/erlio/vernemq/compare/master...dcy:fix_change_nodename

File: apps/vmq_server/src/vmq_subscriber_db.erl [hunk 1]
```diff
@@ -59,17 +59,23 @@ subscribe_db_events() ->
...
-            {Removed, Added} = vmq_subscriber:get_changes(OldSubs, NewSubs),
+            {Removed, Added} = vmq_subscriber:get_changes(get_right_subs(OldSubs), get_right_subs(NewSubs)),
             {update, SubscriberId, Removed, Added};
         (_) ->
             ignore
     end.
 
+get_right_subs(Subs) when is_list(Subs) ->
+    Subs;
+get_right_subs(Subs) ->
```
Comment (dergraf) [inline apps/vmq_server/src/vmq_subscriber_db.erl]: Can you explain the scenario in which this case happen?

Comment (dcy) [inline apps/vmq_server/src/vmq_subscriber_db.erl]: After changed nodename, the old client's Subs will be  ``` {[{'VerneMQ@192.168.3.236',false,[{[<<"chat">>,<<"100">>],1},{[<<"notify">>,<<"100">>],1},{[<<"system">>,<<"100">>],1}]}],['VerneMQ@127.0.0.1']}```

Comment (dcy) [inline apps/vmq_server/src/vmq_subscriber_db.erl]: That is why i asked you why subs will be tuple and list in defferent scenario. https://github.com/erlio/vernemq/issues/241#issuecomment-265946574

Comment (dergraf) [inline apps/vmq_server/src/vmq_subscriber_db.erl]: This is what I get on my machine with 0.15.2 which is a list and not a tuple.
```erlang
{application_start_failure,vmq_server,
  {{undef,[{vmq_subscriber,change_node_all,
    [[{'VerneMQ1@127.0.0.1',false,[{[<<\"a\">>,<<\"b\">>,<<\"+\">>],1}]}],
    'VerneMQ2@127.0.0.1'],[]},{vmq_server_sup,'-maybe_change_nodename/0-fun-2-',3,
...
```
Is it possible that the same node has 'old' subscriptions in the old format?

Comment (dcy) [inline apps/vmq_server/src/vmq_subscriber_db.erl]: after changed ```vmq_subscriber:change_node_all/3```, then login the same client.

Comment (dcy) [inline apps/vmq_server/src/vmq_subscriber_db.erl]:
```
2016-12-09 19:45:06.406 [error] <0.163.0> CRASH REPORT Process <0.163.0> with 0 neighbours exited with reason: {{shutdown,{failed_to_start_child,vmq_queue_sup_sup,{init_error,{function_clause,[{lists,foldl,[#Fun<vmq_subscriber.1.80080411>,[],{[{'VerneMQ1@192.168.3.236',false,[{[<<"chat">>,<<"100">>],1},{[<<"notify">>,<<"100">>],1},{[<<"system">>,<<"100">>],1},{[<<"system">>,<<"all">>],1}]}],['VerneMQ@192.168.3.236']}],[{file,"lists.erl"},{line,1261}]},{vmq_reg,'-fold_subscribers/3-fun-0-',4,[{file,"/home/dcy/my/vernemq/_build/default/lib/vmq_server/src/vmq_reg.erl"},{line,596}]},{plumtree_metadata,fold_it,...}]}}}},...} in application_master:init/4 line 134
```

**Comment by dergraf** [comment]
Thanks @dcy!

**Pull request merged**